src/shared/sol-util-linux.c

```diff
@@ -45,3 +59,107 @@ sol_util_timespec_get_realtime(struct timespec *t)
 {
     return clock_gettime(CLOCK_REALTIME, t);
 }
+
+static int
+rand_gen(struct sol_uuid *p)
+{
+    struct sol_buffer buffer = SOL_BUFFER_INIT_EMPTY;
+    int n = sizeof(*p);
+    int fd = -1;
+    int r;
+
+#ifdef HAVE_GETRANDOM
+    r = syscall(SYS_getrandom, p, n, 0);
+    if (r == n)
+        return 0;
+
+    if (r < 0) {
+        if (errno != EAGAIN) { /* fallback to /dev/urandom on EAGAIN
```

Comment:
Ok, so checking for ENOSYS instead (to continue) is fair enough? I'll remove the comment.